resources/views/pages-legacy/userInfo.blade.php

```diff
@@ -23,7 +23,7 @@
     abort(400);
 }
 
-$userPageModel = User::firstWhere('User', $userPage);
+$userPageModel = User::where('User', $userPage)->orWhere('display_name', $userPage)->first();
```

Comment:
This doesn't actually work. `getUserPageInfo` on the next line down tries to fetch the model where `User=$userPage`, and 404s.

Reply:
This does seem to work if the `display_name` doesn't have special chars:
![Screenshot 2024-07-20 at 10 19 55 AM](https://github.com/user-attachments/assets/74157df8-7e6a-4731-907d-cce8c30b7611)

However, I am able to repro the issue if I try a name that contains non-ASCII chars. I've pushed some small changes to account for this.

We're going to probably need to fix the validation issue at both the code (`isValidUsername`) and the DB level. There are some crazy `display_name` values currently stored in the DB that I'm going to go out on a limb and say we should not support:

* ᜇᜒᜌᜒᜆ᜔ᜇᜓ
* COSTA༬
* Exe³³³
* ??ØØ?

In latest, I've updated `isValidUsername` to add what I think is a reasonable regex-based validation pattern. We can query against this pattern directly in the DB with:
```sql
SELECT * FROM UserAccounts
WHERE display_name NOT REGEXP '^[[:alpha:][:digit:][:space:]_.-]+$';
```

On my local, this yields 6 results. We should remove the display name for all of those users via:
```sql
DELETE FROM UserAccounts
WHERE display_name NOT REGEXP '^[[:alpha:][:digit:][:space:]_.-]+$';
```

Reply:
It works for `Scotty` because there's actually a `Scotty` user, in which case, I'd expect _that_ profile to load.

Given the possible overlap of display names and username, I think you should abandon trying to support display names in the URL.

Reply:
Fair enough - reverted URL support changes in latest. I've also updated the SQL query in the OP to match your suggestion:
```sql
UPDATE `UserAccounts`
SET `display_name` = `User`;
```

Reply:
Long term, I think we want the URL to _always_ look up by display name, since that's the only thing that should be public. If I change my display_name to "UniqueDisplayName", I wouldn't expect anyone to know to go to `/user/Jamiras` to find me.

Reply:
It's a conversation for another time, but we should also maintain some sort of "retired" display_name list. If a user changes their display_name after some period of time (maybe six months or a year), the old display_name gets reserved so someone else wouldn't be able to impersonate the original user.

Reply:
> Long term, I think we want the URL to always look up by display name, since that's the only thing that should be public. If I change my display_name to "UniqueDisplayName", I wouldn't expect anyone to know to go to /user/Jamiras to find me.

That makes sense to me. If we're in a world where all `{user}` routes are looking up by `display_name` instead of `User`, I'm guessing we probably don't need to worry about enforcing no overlap between `display_name` and the username.

> we should also maintain some sort of "retired" display_name list

My understanding is this is what the `user_usernames` table is for.